Extension/src/LanguageServer/configurations.ts

```diff
@@ -739,29 +741,47 @@ export class CppProperties {
         const result: string[] = [];
         if (paths) {
             paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
-            return result;
+            if (paths && !glob) {
+                paths.forEach(entry => {
+                    const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+                    result.push(...entries);
+                });
+                return result;
+            }
         }
 
         const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
-            }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
-            // fastGlob silently strip non-found paths. limit that behavior to dynamic paths only
-            const matches: string[] = fastGlob.isDynamicPattern(normalized) ?
-                fastGlob.sync(normalized, { onlyDirectories: true, cwd}) : [res];
-            globResult.push(...matches.map(s => recursive ? s + '**' : s));
+        if (paths) {
+            for (let res of paths) {
+                let counter: integer = 0;
+                let replacementString: string = '';
+                const lastIndex: integer = res.length - 1;
+                // Detect and glob all wildcard variations by looking at last character in the path first.
+                for (let i: integer = lastIndex; i >= 0; i--) {
+                    if (res[i] === '*') {
+                        counter++;
+                        replacementString = replacementString.replace (/^/, res[i]);
```

Comment:
Instead of building `replacementString` character by character, do it outside the for loop with a slice.
```ts
const suffix = res.slice(res.length - counter - 1);  // not tested, could be off by 1...
```

Reply:
This doesn't work because this will build the string backwards

Reply:
what?  It should trim off the last `counter` characters.

Reply:
Ohh I see

Reply:
Is there a benefit to building the string outside of the for loop? building it outside causes other bugs. I don't see an issue with the way it is now

Reply:
It should give you the same result.  Let's look at it together.